overlord/snapstate/snapmgr.go

```diff
@@ -79,6 +79,8 @@ type SnapSetup struct {
 	Flags SnapSetupFlags `json:"flags,omitempty"`
 
 	SnapPath string `json:"snap-path,omitempty"`
+
+	DownloadInfo snap.DownloadInfo `json:"download-info,omitempty"`
```

Comment:
if we make this one a pointer you can just check for nil in the back compat code